src/Features/Core/Portable/EmbeddedLanguages/StackFrame/StackFrameParser.cs

```diff
@@ -0,0 +1,455 @@
+        /// Ex: System.String[] s
+        ///     ^--------------^ -- Type = "System.String[]"
+        ///                     ^-- Identifier = "s"    
+        /// </summary>
+        private ParseResult<StackFrameParameterDeclarationNode> ParseParameterNode()
+        {
+            var (success, typeIdentifier) = TryParseNameNode(scanAtTrivia: false);
+            if (!success || typeIdentifier is null)
+            {
+                return ParseResult<StackFrameParameterDeclarationNode>.Abort;
+            }
+
+            if (CurrentCharAsToken().Kind == StackFrameKind.OpenBracketToken)
+            {
+                (success, var arrayIdentifiers) = ParseArrayRankSpecifiers();
+                if (!success || arrayIdentifiers.IsDefault)
+                {
+                    return ParseResult<StackFrameParameterDeclarationNode>.Abort;
+                }
+
+                typeIdentifier = new StackFrameArrayTypeExpression(typeIdentifier, arrayIdentifiers);
+            }
+
+            var identifier = _lexer.TryScanIdentifier(scanAtTrivia: false, scanLeadingWhitespace: true, scanTrailingWhitespace: true);
+            if (!identifier.HasValue)
+            {
+                return ParseResult<StackFrameParameterDeclarationNode>.Abort;
+            }
+
+            return new(new(typeIdentifier, identifier.Value));
+        }
+
+        /// <summary>
+        /// Parses the array rank specifiers for an identifier. 
+        /// Ex: string[,][]
+        ///           ^----^ both are array rank specifiers
+        ///                  0: "[,]
+        ///                  1: "[]"
+        /// </summary>
+        private ParseResult<ImmutableArray<StackFrameArrayRankSpecifier>> ParseArrayRankSpecifiers()
+        {
+            using var _ = ArrayBuilder<StackFrameArrayRankSpecifier>.GetInstance(out var builder);
+            using var _1 = ArrayBuilder<StackFrameToken>.GetInstance(out var commaBuilder);
+
+            while (true)
+            {
+                if (!_lexer.ScanCurrentCharAsTokenIfMatch(StackFrameKind.OpenBracketToken, scanTrailingWhitespace: true, out var openBracket))
+                {
+                    return new(builder.ToImmutable());
+                }
+
+                while (_lexer.ScanCurrentCharAsTokenIfMatch(StackFrameKind.CommaToken, scanTrailingWhitespace: true, out var commaToken))
+                {
+                    commaBuilder.Add(commaToken);
+                }
+
+                if (!_lexer.ScanCurrentCharAsTokenIfMatch(StackFrameKind.CloseBracketToken, scanTrailingWhitespace: true, out var closeBracket))
+                {
+                    return ParseResult<ImmutableArray<StackFrameArrayRankSpecifier>>.Abort;
+                }
+
+                builder.Add(new StackFrameArrayRankSpecifier(openBracket, closeBracket, commaBuilder.ToImmutableAndClear()));
+            }
+
+            throw ExceptionUtilities.Unreachable;
```

Comment:
i'm surprised you can even write this line of code.  jsut remove.  the compiler already ensures this is unreachable (since nothing breaks out of the loop ever).